connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -81,13 +82,50 @@ message Relation {
     // Catalog API (experimental / unstable)
     Catalog catalog = 200;
 
+    // ML relation
+    MlRelation ml_relation = 300;
+
     // This field is used to mark extensions to the protocol. When plugins generate arbitrary
     // relations they can add them here. During the planning the correct resolution is done.
     google.protobuf.Any extension = 998;
     Unknown unknown = 999;
   }
 }
 
+message MlRelation {
+  oneof ml_relation_type {
+    ModelTransform model_transform = 1;
+    FeatureTransform feature_transform = 2;
+    ModelAttr model_attr = 3;
+    ModelSummaryAttr model_summary_attr = 4;
+  }
+  message ModelTransform {
+    Relation input = 1;
+    int64 model_ref_id = 2;
+    Params params = 3;
+  }
+  message FeatureTransform {
```

Comment:
happen to find a special case: `Bucketizer`

it is the model of estimator `QuantileDiscretizer`;
itself can be treated as a normal transformer with coefficients set via `setSplits`

Reply:
for a trained `Bucketizer ` we must copy its params from the server, since its param `splits` contains the trained coefficients

Reply:
Got it. Then we can treat it as normal transformer because it only contains splits param